lib/server/server-google-pubsub.ts

```diff
@@ -137,54 +137,98 @@ export class GooglePubSubTransport extends Server implements CustomTransportStra
      * @param pattern - The pattern from the \@GooglePubSubMessageHandler decorator
      */
     private async getSubscriptionFromPattern(pattern: string): Promise<void> {
-        let metadata: GooglePubSubPatternMetadata | null;
+        const metadata = this.parsePattern(pattern);
+
+        const subscriptionName: string = this.getSubscriptionName(metadata, pattern);
+
+        const subscription: GooglePubSubSubscription | null = await this.getSubscription(
+            subscriptionName,
+            metadata.topicName,
+            pattern,
+        );
+
+        if (subscription) {
+            this.logger.log(`Mapped {${subscription.name}} handler`);
+            this.subscriptions.set(pattern, subscription);
+        }
+    }
+
+    /**
+     * Parse a metadata pattern, throwing an exception if it cannot be parsed.
+     *
+     * @throws InvalidPatternMetadataException
+     */
+    private parsePattern = (pattern: string): GooglePubSubPatternMetadata => {
         try {
-            metadata = JSON.parse(pattern);
+            return JSON.parse(pattern);
         } catch (error) {
-            metadata = null;
+            throw new InvalidPatternMetadataException(pattern);
         }
+    };
+
+    /**
+     * Get the name for the subscription based on the given metadata.
+     *
+     * @throws InvalidPatternMetadataException
+     * This exception is thrown if a subscription name cannot be generated.
+     */
+    private getSubscriptionName = (
+        metadata: GooglePubSubPatternMetadata,
+        pattern: string,
+    ): string => {
+        const _topicName = metadata.topicName;
+        const _subscriptionName = metadata.subscriptionName;
 
-        if (metadata == null || !(metadata.subscriptionName || metadata.topicName)) {
+        const subscriptionName: string | null = _topicName
+            ? this.subscriptionNamingStrategy.generateSubscriptionName(
```

Comment:
Do we want to automatically overwrite the subscription name if a topic name is provided? I'm thinking that a subscription is either _always_ passed through the generator or only if only a topic is provided.

Reply:
Thinking on this some more - I think the behavior should be that the subscription name and/or topic name are both computed before we do any action with them. The basic strategies can just pass the names through if they're already set, but a supplied strategy could alter them.

Reply:
That sounds good to me. Makes sense. Definitely the simplest behavior. I'll implement that.